### ISSUE TITLE
NameError: name 'pfam_dir' is not defined

### ISSUE THREAD

**Issue by tbmonica84** [nselem/bigscape#6]
Hello!

I've been trying to run bigscape using conda and I keep getting this error. I'm sure the directory name spelling is correct and  I ran the hmpress command and I can see the Pfam-A.hmm.h3f, .h3i, .h3m and .h3p files, I even try the chmod +x command in case it was a permission issue. I tried with the Pfam version 32 and 34 and I got the same error.
Could you please help me figuring out what's going on?
Please?

Thanks in advance!
Monica

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/monicatorresbeltran/BiG-SCAPE-master/bigscape.py", line 2821, in <module>
    launch_hmmalign(cores, domain_sequence_list)
  File "/Users/monicatorresbeltran/BiG-SCAPE-master/bigscape.py", line 489, in _wrap
    ret = funct(*args)
  File "/Users/monicatorresbeltran/BiG-SCAPE-master/bigscape.py", line 1215, in launch_hmmalign
    pool.map(run_hmmalign, domain_sequence_list)
  File "/Users/monicatorresbeltran/miniconda3/envs/bigscape/lib/python3.10/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/Users/monicatorresbeltran/miniconda3/envs/bigscape/lib/python3.10/multiprocessing/pool.py", line 771, in get
    raise self._value
NameError: name 'pfam_dir' is not defined

**Comment by jorgecnavarrom** [comment]
Mmh, this one's kind of weird because the `pfam_dir` variable is supposed to have a default value (the same folder where the code is), so it should always be defined...
Just to discard quick things: does the folder with the pfam files has a space in its path?

**Comment by tbmonica84** [comment]
Thank you so much for your response.

No, there is no space in the path for the pfam files as they are in the same directory as the code (not in a separarte folder inside the code's directory).

I also check for any misspell and couldn't find any, to double check  I dragged the folder icon into Terminal so that the path  remains as it is

Thank you!!

I'll stay tuned

**Comment by tbmonica84** [comment]
I just re-run it and I'm not sure if the error has to do with the following (it is the warning and error message that comes before the Error I mentioned in my first message, just in case it helps):

warnings.warn(
/Users/monicatorresbeltran/miniconda3/envs/bigscape/lib/python3.10/site-packages/Bio/SubsMat/__init__.py:126: BiopythonDeprecationWarning: Bio.SubsMat has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.substitution_matrices as a replacement, and contact the Biopython developers if you still need the Bio.SubsMat module.
  warnings.warn(
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/Users/monicatorresbeltran/miniconda3/envs/bigscape/lib/python3.10/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/Users/monicatorresbeltran/miniconda3/envs/bigscape/lib/python3.10/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/Users/monicatorresbeltran/BiG-SCAPE-master/bigscape.py", line 1223, in run_hmmalign
    hmmfetch_pars = ["hmmfetch", os.path.join(pfam_dir,"Pfam-A.hmm.h3m"), domain_base]
NameError: name 'pfam_dir' is not defined
"""

**Comment by jorgecnavarrom** [comment]
No, I think that is not related (just a warning for a very unusual case during distance calculation)
On the other hand, how did you install the dependencies? I wonder if the version of any of the packages is related to the error... The easiest here would be to have conda installed and then use the included environment file (i.e. `conda env create -f environment.yml`)

**Comment by tbmonica84** [comment]
Thank you for your prompt reply!

I installed it using using conda, although I didn't use the included environmental file. 

I did:

cd  BiG-SCAPE-master/

conda create --name bigscape

source activate bigscape

conda install numpy scipy scikit-learn
conda install -c bioconda hmmer biopython fasttree
conda install -c anaconda networkx

Do you recommend I start over and install it using the environment file?

Thanks!!

**Comment by tbmonica84** [comment]
I reinstall everything using the environment file as you suggested. Now is running . Thank you so much for your help and advice!

**Comment by jorgecnavarrom** [comment]
Great to hear! I guess a version of some package(s) is causing the trouble
